### ISSUE TITLE
Logging to MLFlow

### ISSUE THREAD

**Issue by sbhavani** [NVIDIA-NeMo/Megatron-Bridge#1387]
**Is your feature request related to a problem? Please describe.**
I need to use MLFlow for experiment tracking and logging in Megatron Bridge training runs. Currently, Megatron Bridge only supports TensorBoard and wandb for logging metrics, hyperparameters, and artifacts. 

**Describe the solution you'd like**
Add native MLFlow logging support to Megatron Bridge's `LoggerConfig` and training, following the same pattern as the existing wandb integration. This would include:
  1. Configuration options in `LoggerConfig`
  2. Logger initialization that:
    - Gets an MLFlow experiment and starts a run with specified name and tags
    - Handles distributed training (e.g. logging only from last rank)
  3. Metric logging throughout the training loop for all metrics currently logged to wandb
  4. Artifact logging for checkpoints

**Describe alternatives you've considered**
- Implementing a general logger plugin: A more extensible solution would be to create a plugin for loggers (e.g. callbacks). This would allow users to implement any logger without modifying core code. This is a larger change that could be considered separately.

**Additional context**

**Comment by snowmanwwg** [comment]
@therealnaveenkamal Are you interested? Super appreciate for all your contributions and help!

**Comment by therealnaveenkamal** [comment]
Sure. I'll take this up @snowmanwwg 

Also, if there are any other interesting non-mlflow issues too, please tag me. I'd be happy to help. Thanks!

**Comment by Phlip79** [comment]
Thank you @therealnaveenkamal! If you need any help at any stage, I am at your disposal.

**Comment by therealnaveenkamal** [comment]
@Phlip79 quick update : i've integrated the mlflow logger. please let me know if you want me to add anything...i'll raise a PR by end of today. Thanks.



I'm recording all the metrics that wandb captures (few are shown here):

<img width="1024" height="668" alt="Image" src="https://github.com/user-attachments/assets/d87d2c73-d2d9-4c48-aaa0-584cbdb2343e" />

As well as the artifacts:

<img width="1024" height="606" alt="Image" src="https://github.com/user-attachments/assets/9cef62fc-f7ca-49b7-a7bc-f2d9850b771f" />

**Comment by Phlip79** [comment]
Thank you @therealnaveenkamal! Looking forward to seeing the PR.

Out of curiosity, where are you hosting the MLFlow instance

**Comment by therealnaveenkamal** [comment]
@Phlip79 right now I'm just using a local instance to run the mlflow setup, but it can be hosted using `mlflow server` command too.
